core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -86,15 +87,15 @@ class BillingProfileManagerDAOImpl(samDAO: SamDAO,
     }
 
     for {
-      billingProfiles <- getAllBillingProfiles(azureConfig, userInfo)
+      billingProfiles <- getAllBillingProfiles(azureConfig, userInfo, rawlsBillingProjects)
     } yield {
       billingProfiles.filter {
         bp => samUserResources.map(_.resourceId).contains(bp.projectName.value)
       }
     }
   }
 
-  private def getAllBillingProfiles(azureConfig: AzureConfig, userInfo: UserInfo)(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]] = {
+  private def getAllBillingProfiles(azureConfig: AzureConfig, userInfo: UserInfo, rawlsBillingProjects: Seq[RawlsBillingProject])(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]] = {
```

Comment:
I think I fouled up the abstraction here during the initial implementation of this DAO. I think this class's sole concern should be BPM model objects (i.e, Profile) and should have little (no?) knowledge of `RawlsBililngProject`s. I think this stitching might better live in the BillingProjectOrchestrator class.

Reply:
I'm trying to understand the scope of what you think might move into `BillingProjectOrchestrator`, which currently only is used for billing project creation. You are suggesting moving `listBillingProjectsV2` to the orchestrator?

Reply:
@aherbst-broad I made an attempt at switching the DAO to return BPM model objects. The only issue I ran into is that for the hard-coded Azure billing project, I really need to return a RawlsBillingProject because I need to include the Rawls billing project name in the return value. Therefore I created a specific method for getting that hard-coded account, and we will just delete that method when we no longer need the hack.

Reply:
(I didn't move anything to `BillingProjectOrchestrator` though)